CI: Java modules tests are broken

### Describe the bug

The modules are broken since they are trying to connect to the locally created cluster instead of the provisioned memorydb

This happens as a result of https://github.com/valkey-io/valkey-glide/pull/2352 which introduces steps (beforeTests) the tests, breaking the assumptions or exposing a different issue with the interoperability between
```./gradlew :integTest:modulesTest -Dcluster-endpoints=*** -Dtls=true```
and running startStandalone, startCluster and startClusterForAz

The result seems to be overwritting the modules endpoints provided with ```-Dcluster-endpoints=***```

Can be seen in the Java CI workflow output for main branch:

```Run ./gradlew :integTest:modulesTest -Dcluster-endpoints=*** -Dtls=true
  
Starting a Gradle Daemon, 36 busy and 4 incompatible and 2 stopped Daemons could not be reused, use --status for details
> Task :integTest:stopAllBeforeTests
INFO:root:## Executing cluster_manager.py with the following args:
> Task :integTest:clearDirs
  Namespace(host='12[7](https://github.com/valkey-io/valkey-glide/actions/runs/13696338647/job/38299507428?pr=3340#step:7:7).0.0.1', tls=False, auth=None, log='info', logfile=None, action='stop', folder_path='/home/ubuntu/action-runner-ilia/_work/valkey-glide/valkey-glide/utils/clusters', prefix='cluster', cluster_folder=None, keep_folder=False, pids='')
INFO:root:2025-03-06 10:1[7](https://github.com/valkey-io/valkey-glide/actions/runs/13696338647/job/38299507428?pr=3340#step:7:8):31.608125+00:00 Stopping script for cluster/s cluster* in /home/ubuntu/action-runner-ilia/_work/valkey-glide/valkey-glide/utils/clusters
INFO:root:Cluster stopped in 0.0002 seconds
> Task :integTest:startStandalone
INFO:root:## Executing cluster_manager.py with the following args:
  Namespace(host='127.0.0.1', tls=True, auth=None, log='info', logfile=None, action='start', cluster_mode=False, folder_path='/home/ubuntu/action-runner-ilia/_work/valkey-glide/valkey-glide/utils/clusters', ports=None, shard_count=3, replica_count=0, prefix='cluster', load_module=None)
INFO:root:2025-03-06 10:17:31.7[9](https://github.com/valkey-io/valkey-glide/actions/runs/13696338647/job/38299507428?pr=3340#step:7:10)9346+00:00 Starting script for cluster /home/ubuntu/action-runner-ilia/_work/valkey-glide/valkey-glide/utils/clusters/tls-cluster-2025-03-06T[10](https://github.com/valkey-io/valkey-glide/actions/runs/13696338647/job/38299507428?pr=3340#step:7:11)-17-31Z-Ou9GRQ
Traceback (most recent call last):
  File "/home/ubuntu/action-runner-ilia/_work/valkey-glide/valkey-glide/utils/cluster_manager.py", line 1256, in <module>
    main()
  File "/home/ubuntu/action-runner-ilia/_work/valkey-glide/valkey-glide/utils/cluster_manager.py", line [11](https://github.com/valkey-io/valkey-glide/actions/runs/13696338647/job/38299507428?pr=3340#step:7:12)92, in main
    servers = create_servers(
  File "/home/ubuntu/action-runner-ilia/_work/valkey-glide/valkey-glide/utils/cluster_manager.py", line 446, in create_servers
    generate_tls_certs()
  File "/home/ubuntu/action-runner-ilia/_work/valkey-glide/valkey-glide/utils/cluster_manager.py", line 131, in generate_tls_certs
    make_key(ca_key, 4096)
  File "/home/ubuntu/action-runner-ilia/_work/valkey-glide/valkey-glide/utils/cluster_manager.py", line [12](https://github.com/valkey-io/valkey-glide/actions/runs/13696338647/job/38299507428?pr=3340#step:7:13)4, in make_key
    output, err = p.communicate(timeout=10)
  File "/usr/lib/python3.10/subprocess.py", line 1154, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
  File "/usr/lib/python3.10/subprocess.py", line 2022, in _communicate
    self._check_timeout(endtime, orig_timeout, stdout, stderr)
  File "/usr/lib/python3.10/subprocess.py", line 1198, in _check_timeout
    raise TimeoutExpired(
subprocess.TimeoutExpired: Command '['openssl', 'genrsa', '-out', '/home/ubuntu/action-runner-ilia/_work/valkey-glide/valkey-glide/utils/tls_crts/ca.key', '4096']' timed out after 10 seconds
FAILURE: Build failed with an exception.
* Where:
Build file '/home/ubuntu/action-runner-ilia/_work/valkey-glide/valkey-glide/java/integTest/build.gradle' line: 115
* What went wrong:
Execution failed for task ':integTest:startStandalone'.
> Process 'command 'python3'' finished with non-zero exit value 1
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
> Task :integTest:startStandalone FAILED
> Task :integTest:startClusterForAz
> Task :integTest:startCluster
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
BUILD FAILED in [21](https://github.com/valkey-io/valkey-glide/actions/runs/13696338647/job/38299507428?pr=3340#step:7:22)s
For more on this, please refer to https://docs.gradle.org/8.5/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
5 actionable tasks: 5 executed
Error: Process completed with exit code 1
```


### Expected Behavior

Modules test pass

### Current Behavior

Modules tests fail

### Reproduction Steps

Trigger Java CI modules tests on main

### Possible Solution

revert or fix the new steps

### Additional Information/Context

_No response_

### Client version used

main

### Engine type and version

dont care

### OS

ubuntu

### Language

Java

### Language Version

dont care

### Cluster information

_No response_

### Logs

_No response_

### Other information

_No response_

--- a/java/integTest/build.gradle
+++ b/java/integTest/build.gradle
@@ -137,8 +137,10 @@ compileTestJava.dependsOn ':client:publishToMavenLocal'
 
 tasks.withType(Test) {
     useJUnitPlatform()
-    dependsOn 'beforeTests'
-    finalizedBy 'afterTests'
+    if (!project.gradle.startParameter.taskNames.contains(':integTest:modulesTest')) {
+        dependsOn 'beforeTests'
+        finalizedBy 'afterTests'
+    }
 
     doFirst {
         systemProperty 'test.server.standalone', standaloneHosts

--- a/java/integTest/src/test/java/glide/TestConfiguration.java
+++ b/java/integTest/src/test/java/glide/TestConfiguration.java
@@ -9,9 +9,9 @@ import glide.api.BaseClient;
 import glide.api.GlideClient;
 import glide.api.GlideClusterClient;
 import glide.api.logging.Logger;
+import org.apache.commons.lang3.tuple.Pair;
 
 public final class TestConfiguration {
-    // All servers are hosted on localhost
     public static final String[] STANDALONE_HOSTS =
             System.getProperty("test.server.standalone", "").split(",");
     public static final String[] CLUSTER_HOSTS =
@@ -24,21 +24,64 @@ public final class TestConfiguration {
     static {
         Logger.init(Logger.Level.OFF);
         Logger.setLoggerConfig(Logger.Level.OFF);
+
+        System.out.printf("STANDALONE_HOSTS = %s\n", System.getProperty("test.server.standalone", ""));
+        System.out.printf("CLUSTER_HOSTS = %s\n", System.getProperty("test.server.cluster", ""));
+        System.out.printf("AZ_CLUSTER_HOSTS = %s\n", System.getProperty("test.server.azcluster", ""));
+
+        var result = getVersionFromStandalone();
+        if (result.getKey() != null) {
+            SERVER_VERSION = result.getKey();
+        } else {
+            var errorStandalone = result.getValue();
+            result = getVersionFromCluster();
+            if (result.getKey() != null) {
+                SERVER_VERSION = result.getKey();
+            } else {
+                var errorCluster = result.getValue();
+                errorStandalone.printStackTrace(System.err);
+                System.err.println();
+                errorCluster.printStackTrace(System.err);
+                throw new RuntimeException("Failed to get server version");
+            }
+        }
+        System.out.printf("SERVER_VERSION = %s\n", SERVER_VERSION);
+    }
+
+    private static Pair<Semver, Exception> getVersionFromStandalone() {
+        if (STANDALONE_HOSTS[0].isEmpty()) {
+            return Pair.of(null, new Exception("No standalone nodes given"));
+        }
         try {
-            BaseClient client =
-                    !STANDALONE_HOSTS[0].isEmpty()
-                            ? GlideClient.createClient(commonClientConfig().build()).get()
-                            : GlideClusterClient.createClient(commonClusterClientConfig().build()).get();
+            BaseClient client = GlideClient.createClient(commonClientConfig().build()).get();
 
             String serverVersion = TestUtilities.getServerVersion(client);
             if (serverVersion != null) {
-                SERVER_VERSION = new Semver(serverVersion);
+                return Pair.of(new Semver(serverVersion), null);
             } else {
-                throw new Exception("Failed to get server version");
+                return Pair.of(null, new Exception("Failed to parse version"));
             }
-
         } catch (Exception e) {
-            throw new RuntimeException("Failed to get server version", e);
+            return Pair.of(null, e);
+        }
+    }
+
+    private static Pair<Semver, Exception> getVersionFromCluster() {
+        if (CLUSTER_HOSTS[0].isEmpty()) {
+            return Pair.of(null, new Exception("No cluster nodes given"));
+        }
+        try {
+            BaseClient client =
+                    GlideClusterClient.createClient(commonClusterClientConfig().build()).get();
+
+            String serverVersion = TestUtilities.getServerVersion(client);
+            if (serverVersion != null) {
+                return Pair.of(new Semver(serverVersion), null);
+            } else {
+                return Pair.of(null, new Exception("Failed to parse version"));
+            }
+        } catch (Exception e) {
+            return Pair.of(null, e);
         }
     }
 }

